### PR TITLE
jellyfish db datatype

### PR DESCRIPTION
**Pull request by astrovsky01**
## What did you do? 
- [describe the proposed changes]
- add jellyfish db datatype

## Why did you make this change?
(Cite Issue number OR provide rationalization of changes if no issue exists)
(If fixing a bug, please add any relevant error or traceback)
Adding Jellyfish tool to IUC (PR for IUC pending)

## How to test the changes? 
(select the most appropriate option; if the latter, provide steps for testing below)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

## For UI Components
- [ ] I've included a screenshot of the changes

### CODE REVIEW

File: lib/galaxy/config/sample/datatypes_conf.xml.sample [hunk 1]
```diff
@@ -184,6 +184,7 @@
       <display file="rviewer/bed.xml" inherit="true"/>
       <display file="igv/interval_as_bed.xml" inherit="true"/>
     </datatype>
+    <datatype extension="jf" type="galaxy.datatypes.binary:Binary" subclass="true" display_in_upload="true" description="Jellyfish database files are k-mer counts in binary format with a readable head. They are operated on and converted to human-readable text through jellyfish commands." />
```
Comment (bgruening) [inline lib/galaxy/config/sample/datatypes_conf.xml.sample]: I would call the extension `jellyfish`. Two letters are really not that unique and meaningful.

**Comment by bgruening** [comment]
You probably need this on ORG? Do you want to target 21.01?

File: lib/galaxy/config/sample/datatypes_conf.xml.sample (hunk 1 above)
Comment (astrovsky01) [reply lib/galaxy/config/sample/datatypes_conf.xml.sample]: Good idea

**Comment by astrovsky01** [comment]
@bgruening good idea on the target

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
The failing test looks unrelated.

**Pull request merged**